fast64_internal/sm64/sm64_geolayout_writer.py

```diff
@@ -1037,179 +1059,149 @@ def geoWriteTextDump(textDumpFilePath, geolayoutGraph, levelData):
-                            optionGeolayout,
-                            geolayoutGraph,
-                            optionGeolayout.name,
-                        )
-                if material is not None:
-                    overrideNodes.append(copyNode)
-            i += 1
-    for i in range(len(transformNode.children)):
-        childNode = transformNode.children[i]
-        if isinstance(transformNode.node, SwitchNode):
-            prefixName = prefix + "_opt" + str(i)
+            prefix_name = f"{prefix}_opt{child_index}"
+            child_node_tex_dimensions = child_node.node.texDimensions
+            if (
+                child_node_tex_dimensions is not None
+                and materialOverrideTexDimensions is not None
+                and materialOverrideTexDimensions != tuple(child_node_tex_dimensions)
+            ):
+                raise PluginError(
+                    'In switch bone "'
+                    + switchName
+                    + '", some material '
+                    + "overrides \nhave textures with dimensions differing from the original material.\n"
+                    + "UV coordinates are in pixel units, so there will be UV errors in those overrides.\n "
+                    + "Make sure that all overrides have the same texture dimensions as the original material.\n"
+                    + "Note that materials with no textures default to dimensions of 32x32."
+                )
+
+            if child_node_tex_dimensions is not None:
+                materialOverrideTexDimensions = tuple(child_node_tex_dimensions)
+
+            # Switch option bones should have unique names across all armatures.
+            transform_node.children.remove(child_node)  # remove child node aka switch override node
+            option_geo_layout = geolayoutGraph.addGeolayout(
+                child_node, prefix_name
+            )  # add geo layout starting from child node
+            geolayoutGraph.addJumpNode(
+                transform_node, geolayout, option_geo_layout, child_index
+            )  # replace switch override with jump node to override GL
+
+            # for switch overrides, each one is a copy of the first child, which is the default option, so iterate the first childs
+            # hierarchy and create a copy of that and place it in the option_geo_layout
+            generateOverrideHierarchy(
+                None,
+                transform_node.children[0],
+                child_node.node.material,
+                child_node.node.specificMat,
+                child_node.node.overrideType,
+                child_node.node.drawLayer,
+                0,
+                option_geo_layout,
+                geolayoutGraph,
+                option_geo_layout.name,
+            )
+
+            # if this is a material override, add to the list
+            # if child_node.node.material is not None:
+            # overrideNodes.append(optionGeolayout.nodes[0])
```

Comment:
not sure if this is intended but if it's leftover commented code then it should be removed